commit 92328a7cb33802982722156f99a050438dedfe3d
Author: Gordon Sim <gsim@apache.org>
Date:   Fri Apr 8 09:34:20 2011 +0000

    Bug 693847 - Fixed - JobServer's qmf methods are not callable from qpid-tool
    
    QPID-3190: always try to eval() quoted tokens
    
    git-svn-id: https://svn.apache.org/repos/asf/qpid/trunk@1090170 13f79535-47bb-0310-9956-ffa450edef68
    (cherry picked from commit 95e86a3d725a876d3933b850427f94ad0ac93377)

diff --git a/qpid/tools/src/py/qpid-tool b/qpid/tools/src/py/qpid-tool
index 2a06dae..d6bb9bc 100755
--- a/qpid/tools/src/py/qpid-tool
+++ b/qpid/tools/src/py/qpid-tool
@@ -267,10 +267,10 @@ class QmfData(Console):
       ##
       ## TODO: use a regex for this instead of this convoluted logic,
       ## or even consider passing all args through eval() [which would
-      ## be a minor change to the nterface as string args would then
+      ## be a minor change to the interface as string args would then
       ## always need to be quoted as strings within a map/list would
       ## now]
-      if arg[0] == '{' or arg[0] == '[' or arg == "True" or arg == "False" or \
+      if arg[0] == '{' or arg[0] == '[' or arg[0] == '"' or arg[0] == '\'' or arg == "True" or arg == "False" or \
             ((arg.count('.') < 2 and (arg.count('-') == 0 or \
             (arg.count('-') == 1 and  arg[0] == '-')) and \
             arg.replace('.','').replace('-','').isdigit())):
